tasks/secure-installation.yml

```diff
@@ -1,87 +1,73 @@
 ---
-- name: "Get MariaDB version."
-  ansible.builtin.command: 'mysql --version'
-  register: _mariadb_cli_version_
-  changed_when: false
-  check_mode: false
-
 - name: "Ensure MariaDB daemon is started"
   ansible.builtin.service:
     name: "{{ _mariadb_daemon }}"
     state: started
 
+- name: Copy .my.cnf file with root password credentials.
+  ansible.builtin.template:
+    src: "root-my.cnf.j2"
+    dest: "{{ mariadb_admin_home }}/.my.cnf"
+    owner: root
+    group: root
+    mode: 0600
+  when: mariadb_admin_force_password_update
+
 - name: Disallow root login remotely
-  ansible.builtin.command: mysql -NBe
-    "DELETE FROM mysql.user
-    WHERE User='{{ mariadb_admin_user }}'
-    AND Host NOT IN ('localhost', '127.0.0.1', '::1')"
-  changed_when: false
+  community.mysql.mysql_user:
+    name: "{{ mariadb_admin_user }}"
+    login_unix_socket: "{{ _mariadb_socket }}"
+    host: "%"
+    priv: "*.*:ALL,GRANT"
+    state: absent
```

Comment:
It doesn't do the same thing, you need to disallow login from other hosts, not only `%`